### DISCUSSION TITLE
Cloud messaging - secondaryApp

### DISCUSSION THREAD

**Discussion by iwalker7** [invertase/react-native-firebase#5250] [Q&A] [answered]
Support for multi-project cloud messaging?

**Comment by mikehardy** [comment, marked as answer]
It doesn't support it.
The products that do have constructors that can take a FirebaseApp, like Functions https://firebase.google.com/docs/reference/android/com/google/firebase/functions/FirebaseFunctions#getInstance(com.google.firebase.FirebaseApp)

Messaging doesn't https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessaging#getInstance()

Nothing to do with react-native-firebase, it's a messaging limitation, since the device receives messages and the device is a singular thing, the messages module is a singular thing.

**Reply by iwalker7** [reply]
i mean is it possible [https://firebase.google.com/docs/projects/multiprojects#use_multiple_projects_in_your_application](https://firebase.google.com/docs/projects/multiprojects#use_multiple_projects_in_your_application) in rnMessaging

**Reply by mikehardy** [reply]
No it is not possible.
You quote the *general* docs for the idea. Each module supports it or not. They do not all support it.
Messaging in particular does not, and I gave you the *exact* link to the object APIs that prove the same.

**Reply by iwalker7** [reply]
Okey. thank you
